src/utils/codec.ts

```diff
@@ -0,0 +1,418 @@
+    }
+}
+
+/**
+ * Codec
+ *
+ * Encoding/Decoding string with Unicode code points(0 - 0x10FFFF).
+ *
+ * All data will be formated into binary while constructing Codec
+ * class, for example, Let's format the word `Mask`:
+ *
+ * For `Base1024`:
+ *
+ * | Text |   Binary   | Index |
+ * | ---- | ---------- | ----- |
+ * |      | 0100110101 |  309  |
+ * |      | 1000010111 |  535  |
+ * |      | 0011011011 |  219  |
+ *
+ * For `UTF-16`: `[ 01001101, 01100001, 01110011, 01101011 ]`
+ *
+ * | Text |  Binary  | Index |
+ * | ---- | -------- | ----- |
+ * |  M   | 01001101 |  77   |
+ * |  a   | 01100001 |  97   |
+ * |  s   | 01110011 |  115  |
+ * |  k   | 01101011 |  107  |
+ *
+ * For `Base64`: `[ 010011, 010110, 000101, 110011, 011010, 11 ]`
+ *
+ * | Text | Binary | Index |
+ * | ---- | ------ | ----- |
+ * |  T   | 010011 |  19   |
+ * |  W   | 010110 |  22   |
+ * |  F   | 000101 |  5    |
+ * |  z   | 110011 |  51   |
+ * |  a   | 011010 |  26   |
+ * |  s   | 110000 |  48   |
+ * |  =   | 000000 |       |
+ *
+ * For `hex`: `[ 0100, 1101, 0110, 0001, 0111, 0011, 0110, 1011 ]`
+ *
+ * | Text | Binary | Index |
+ * | ---- | ------ | ----- |
+ * |  4   |  0100  |   4   |
+ * |  d   |  1101  |   13  |
+ * |  6   |  0110  |   6   |
+ * |  1   |  0001  |   1   |
+ * |  7   |  0111  |   7   |
+ * |  3   |  0011  |   3   |
+ * |  6   |  0110  |   6   |
+ * |  b   |  1011  |   11  |
+ *
+ * @property {Point[]} points - Code points
+ *
+ * @method  toHex    - Convert code points to hex string
+ * @method  toBase64 - Convert code points to base64 string
+ * @method  toString - Convert code points to utf-16 string
+ */
+export class Codec {
+    /**
+     * Build `Codec` from target scheme string
+     *
+     * @param  {CodecScheme} scheme - target codec scheme
+     * @param  {string}      input  - the base `CodecScheme` string
+     */
+    static from(scheme: CodecScheme, input: string | string[], emoji: boolean = false): Codec {
+        assertScheme(scheme)
+
+        // Map bytes to binary string
+        let bits: number = 0
+        const mapFn = (byte: string) => {
+            // return empty string if it is tail
+            if (byte === scheme.tail) {
+                bits += 2
+                return ''
+            }
+
+            // throw error if out of range
+            const index = scheme.table.indexOf(byte)
+            if (index === -1) {
+                throw new Error(`Decode from ${scheme.name} string failed`)
+            }
+
+            // Complete
+            return String(('0'.repeat(scheme.bits - 1) + Number(index).toString(2)).slice(-scheme.bits))
+        }
+
+        // Check if need to parse emojis
+        if (emoji) {
+            const tails = String(input.slice(input.indexOf(scheme.tail)))
+            input = parse(input as string).map((emoji: Record<string, any>) => emoji.text)
+            input = input.concat(tails.split(''))
+        }
```

Comment:
`twemoji-parser` doesn't work when we have a really long emojis string. (some emojis can not be parsed by the `parser`, even they listed in `twemoji`...need to verify again)

Reply:
The length of No.1024 emoji in `BASE1024` is `2`, and the length of No.2048 emoji in `BASE2048` is `6`, this makes the `encoding/decoding` inestimable

Or we can write our own emoji parser to solve this, substring the encoding `BASE1024/BASE2048` by our emoji list, this costs a lot but workable, I'll give it a try.